File: 452-minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.ts
function findMinArrowShots(points: number[][]): number {
    points.sort((a,b) => a[0] - b[0]);
    let count = points.length;
    let prev = points[0];
    for(let i = 1; i < points.length; i ++){
        let cur = points[i];
        if(cur[0] <= prev[1]){
            count -= 1;
            prev[0] = cur[0];
            prev[1] = Math.min(cur[1],prev[1]);
        }
        else{
            prev = cur;
        }
    }
    return count;

};

File: 452-minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points,(a,b) -> Integer.compare(a[0], b[0]));
        int mergeCount = 0;
        for(int i = 0; i < points.length - 1; i ++) {
            int[] curInterval = points[i];
            int[] nextInterval = points[i + 1];
            if(curInterval[1] >= nextInterval[0]){
                mergeCount ++;
                nextInterval[0] = Math.max(curInterval[0], nextInterval[0]);
                nextInterval[1] = Math.min(curInterval[1], nextInterval[1]);
            }
        }
        return points.length - mergeCount;
    }
}